File: start/leetcode/subarray-sum-equals-k.ts
function subarraySum(nums: number[], k: number): number {
    
    const map = new Map<number,number>();
    let count: number = 0
    let sum: number = 0
    map.set(0,1)
    
    
    for(let i=0;i<nums.length;i++){
        sum+=nums[i]
        if(map.has(sum-k)){
            count+=map.get(sum-k)
        }
        
        map.set(sum,(map.get(sum) || 0)+1)
        
    }
    
    return count
    
};

File: start/leetcode/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        m={0:1}
        curSum=0
        res=0
        for i in nums:
            curSum+=i
            diff=curSum-k

            res+=m.get(diff,0)
            m[curSum]=m.get(curSum,0)+1
        return res

